### PR TITLE
dev: fix typo in alpine guard check

### PR DESCRIPTION
**Pull request by rvantonder**
Spotted this when I saw a [`Misc Linters` job failing](https://buildkite.com/sourcegraph/sourcegraph/builds/86725#a5b89b64-00d7-4237-a632-72409a88678d/1364-1365) (doesn't affect green build?)

### CODE REVIEW

**Review by efritz — APPROVED** [review]

**Comment by rvantonder** [comment]
Crap, @efritz now what, this is a mess: https://buildkite.com/sourcegraph/sourcegraph/builds/86735#345d4e2e-398d-4051-ac15-ea9217452fc9/701-703

**Comment by efritz** [comment]
@rvantonder I don't understand. That file is inside a `testdata` directory.

**Comment by rvantonder** [comment]
Hmmm, dunno, is this pattern correctly matching the path?

File: dev/check/no-alpine-guard.sh [hunk 1]
```diff
@@ -19,7 +19,7 @@ ALPINE_MATCHES=$(git grep -e '\salpine\:' --and --not -e '^\s*//' --and --not -e
   ':(exclude)doc/campaigns/' \
   ':(exclude)web/src/enterprise/campaigns/create/CreateCampaignPage.tsx' \
   ':(exclude)vendor' \
```
Comment (rvantonder) [inline dev/check/no-alpine-guard.sh:21]:
```suggestion
  ':(exclude)*vendor*' \
```

File: dev/check/no-alpine-guard.sh [hunk 2]
```diff
@@ -18,8 +18,8 @@ ALPINE_MATCHES=$(git grep -e '\salpine\:' --and --not -e '^\s*//' --and --not -e
   ':(exclude)docker-images/alpine' \
   ':(exclude)doc/campaigns/' \
   ':(exclude)web/src/enterprise/campaigns/create/CreateCampaignPage.tsx' \
-  ':(exclude)vendor' \
-  ':(eclude)testdata')
+  ':(exclude)*vendor*' \
+  ':(exclude)testdata')
```
Comment (rvantonder) [inline dev/check/no-alpine-guard.sh:22]:
```suggestion
  ':(exclude)*testdata*')
```

**Comment by rvantonder** [comment]
@efritz that regex wasn't working locally, it needs glob expansion to match a path containing `testdata`. Similarly, the `vendor` exclude was doing nothing (we just happened not to match anything). So I'm changing these and calling it "done".

**Review by efritz — APPROVED** [review]

**Pull request merged**